src/borg/_hashindex.c

```diff
@@ -187,10 +189,16 @@ hashindex_lookup(HashIndex *index, const unsigned char *key, int *start_idx)
             idx -= index->num_buckets;
         }
         if(idx == start) {
+            /* we have done a full pass over all buckets. */
             break;
         }
```

Comment:
As we avoid a full HT, this likely never happens.

But if it would happen:
- the code would avoid an endless loop by breaking out
- in case there was no deleted/tombstone bucket, it would return `start` in `*start_idx` and that bucket would be **full**. if the caller checks the bucket state, the caller might notice, but maybe we should fail quicker and just right here?

Reply:
Good point.  So if didx == -1, abort the program, since this should never happen?  (IIUC, it should also not happen with didx != -1, so we could log a warning in that case?  Or just abort in either case, since it would indicate something seriously wrong?)

Reply:
Could this case happen with one of the `LOAD` variables set to .99 or .999 due to rounding issues?

Reply:
^ made another commit with "fail early".

Reply:
had to force push again to fix wording / typos in commit comment, no other changes.

Reply:
ok, i've implemented all of this.